crypto/des/fcrypt.c

```diff
@@ -70,9 +70,12 @@ char *DES_crypt(const char *buf, const char *salt)
     char e_buf[32 + 1];         /* replace 32 by 8 ? */
     char *ret;
 
+    if (sizeof(salt) < 2) return NULL;
```

Comment:
Don’t do anything; if the caller passes in NULL the code will crash and they will fix things.